_studio/shared/include/libmfx_core_hw.h

```diff
@@ -18,15 +18,54 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-#include "mfx_common.h"
-
-
 #ifndef __LIBMFX_CORE__HW_H__
 #define __LIBMFX_CORE__HW_H__
 
+#include "mfx_common.h"
 #include "umc_va_base.h"
+#include <memory>
+#include <mutex>
+#include <vector>
+
+struct PciDisplayDevice
+{
+    mfxU32 vendor_id;
+    mfxU32 device_id;
+};
+
+class HardwareInfoStore
+{
+public:
+    static HardwareInfoStore& GetInstance()
+    {
+        std::lock_guard<std::mutex> lock(m_mtx);
```

Comment:
@vshampor 
I'd rather use
> static m_instance

in [GetInstance] method scope directly, instead of having it at class scope. No static instance of mutex is required here, initialization of block-scope variables with static storage duration is thread safe since C++11 (see details at [6.7 Declaration statement](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2013/n3797.pdf), c.4).
Profit:
1. U don't need this mutex to be static =>
2. U don't need [libmfx_core_hw.cpp :: std::mutex HardwareInfoStore::m_mtx] definition =>
3. U don't mix mutex usage from static/non-static methods.